nifi-extension-bundles/nifi-mongodb-bundle/nifi-mongodb-services/src/main/java/org/apache/nifi/mongodb/MongoDBControllerService.java

```diff
@@ -96,27 +99,53 @@ protected MongoClient createClient(ConfigurationContext context, MongoClient exi
         }
 
         try {
-            final String uri = context.getProperty(URI).evaluateAttributeExpressions().getValue();
+            String uri = context.getProperty(URI).evaluateAttributeExpressions().getValue();
             final String user = context.getProperty(DB_USER).evaluateAttributeExpressions().getValue();
             final String passw = context.getProperty(DB_PASSWORD).evaluateAttributeExpressions().getValue();
 
             final MongoClientSettings.Builder builder = MongoClientSettings.builder();
-            final ConnectionString cs = new ConnectionString(uri);
 
-            if (user != null && passw != null) {
-                final String database = cs.getDatabase() == null ? "admin" : cs.getDatabase();
+            // Regex to find authMechanism value (case-insensitive)
+            final Pattern authMechanismPattern = Pattern.compile("(?i)(?:[?&])authmechanism=([^&]*)");
```

Comment:
This pattern can be defined in a static variable.

Reply:
Done